maintain/going-production-with-k8s.md

```diff
@@ -0,0 +1,468 @@
+ parameters:
+   repl: "3"
+   priority_io: "high"
+ ```
+  
+* Volumes can be created in different availability zones by using the `--zones` option in the `pxctl volume create` command
+
+For e.g., for PX cluster running on AWS, 
+
+ ```yaml
+ kind: StorageClass
+ apiVersion: storage.k8s.io/v1beta1
+ metadata:
+   name: portworx-sc
+ provisioner: kubernetes.io/portworx-volume
+ parameters:
+   repl: "3"
+   priority_io: "high"
+   zones: "us-east-1a"
+ ```
+* Volumes can be created in different racks using `--racks` option and passing the rack labels when creating the volume
+
+ ```yaml
+ kind: StorageClass
+ apiVersion: storage.k8s.io/v1beta1
+ metadata:
+   name: portworx-sc
+ provisioner: kubernetes.io/portworx-volume
+ parameters:
+   repl: "3"
+   priority_io: "high"
+   racks: "rack1"
+ ```
+
+* If the volumes need to be protected against accidental deletes because of background garbage collecting scripts, 
+  then the volumes need to enabled with `--sticky` flag
+  
+ ```yaml
+ kind: StorageClass
+ apiVersion: storage.k8s.io/v1beta1
+ metadata:
+   name: portworx-sc
+ provisioner: kubernetes.io/portworx-volume
+ parameters:
+   repl: "3"
+   priority_io: "high"
+   racks: "rack1"
+   sticky: "true"
+ ```
+  
+* For applications that require shared access from multiple containers running in different hosts, 
+  Portworx recommends running  shared volumes. Shared volumes can be configured as follows by adding `shared: "true" ` to the storage class:
+ 
+ ```yaml
+ kind: StorageClass
+ apiVersion: storage.k8s.io/v1beta1
+ metadata:
+   name: portworx-sc
+ provisioner: kubernetes.io/portworx-volume
+ parameters:
+   repl: "3"
+   priority_io: "high"
+   racks: "rack1"
+   shared: "true"
+ ```
+
+  This [page](https://docs.portworx.com/manage/volumes.html) gives more details on different volume types, 
+  how to create them and update the configuration for the volumes 
+
+### Data Protection for Containers
+
+* Snapshots - Follow DR best practices and ensure volume snapshots are scheduled for instantaneous recovery in the 
+  case of app failures. 
+  
+* Portworx support 64 snapshots per volume. 
+
+* Refer to this [document](https://docs.portworx.com/manage/snapshots.html) for a brief overview on how to manage snapshots via `pxctl`. In Kubernetes, most snapshot functionality can be handled via kubernetes command line.
+
+* Periodic scheduled snapshots can be setup by defining the `snap_interval` in the Portworx StorageClass. An example is shown below.
+
+```yaml
+kind: StorageClass
+apiVersion: storage.k8s.io/v1beta1
+  metadata:
+    name: portworx-repl-1-snap-internal
+provisioner: kubernetes.io/portworx-volume
+parameters:
+  repl: "1"
+  snap_interval: "240"
+```
+
+* You can use annotations in Kubernetes to perform on-demand snapshot operations from within Kubernetes. 
+
+Portworx uses a special annotation `px/snapshot-source-pvc` which can be used to identify the name of the source PVC whose snapshot needs to be taken.
```

Comment:
Why duplicate this info here instead of just reference the page https://docs.portworx.com/scheduler/kubernetes/snaps.html#using-annotations? It gets difficult to maintain the docs later on if the same steps are referenced at multiple places.

Reply:
Like to keep some examples inline. So it is not with too many links to jump around